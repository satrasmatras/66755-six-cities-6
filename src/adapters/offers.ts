import Offer from "../models/offer";

export const adaptDataToOffer = (data: any): Offer => {

  const offer = {
    ...data,
    previewImage: data[`preview_image`],
    isFavorite: data[`is_favorite`],
    isPremium: data[`is_premium`],
    maxAdults: data[`max_adults`],
    host: {
      ...data.host,
      isPro: data.host[`is_pro`],
      avatarUrl: data.host[`avatar_url`],
    }
  };

  delete offer[`preview_image`];
  delete offer[`is_favorite`];
  delete offer[`is_premium`];
  delete offer[`max_adults`];
  delete offer.host[`is_pro`];
  delete offer.host[`avatar_url`];

  return offer;
};

export const adaptOfferToData = (offer: any): Offer => {

  const data = {
    ...offer,
    previewImage: offer[`preview_image`],
    isFavorite: offer[`is_favorite`],
    isPremium: offer[`is_premium`],
    maxAdults: offer[`max_adults`],
    host: {
      ...offer.host,
      isPro: offer.host[`is_pro`],
      avatarUrl: offer.host[`avatar_url`],
    }
  };

  delete data[`preview_image`];
  delete data[`is_favorite`];
  delete data[`is_premium`];
  delete data[`max_adults`];
  delete data.host[`is_pro`];
  delete data.host[`avatar_url`];

  return data;
};
