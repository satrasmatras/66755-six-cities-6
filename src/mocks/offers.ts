import Offer from "../models/offer";

const OFFERS: Offer[] = [
  {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      },
      "name": `Amsterdam`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "avatarUrl": `img/1.png`,
      "id": 3,
      "isPro": true,
      "name": `Angelina`
    },
    "id": 1,
    "images": [`img/1.png`, `img/2.png`],
    "isFavorite": true,
    "isPremium": true,
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.85309666406198,
      "zoom": 8
    },
    "maxAdults": 4,
    "previewImage": `img/apartment-03.jpg`,
    "price": 150,
    "rating": 4.8,
    "title": `Beautiful & luxurious studio`,
    "type": `apartment`
  },
  {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      },
      "name": `Amsterdam`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "avatarUrl": `img/1.png`,
      "id": 3,
      "isPro": true,
      "name": `Angelina`
    },
    "id": 2,
    "images": [`img/1.png`, `img/2.png`],
    "isFavorite": false,
    "isPremium": false,
    "location": {
      "latitude": 52.369553943508,
      "longitude": 4.85309666406198,
      "zoom": 8
    },
    "maxAdults": 4,
    "previewImage": `img/apartment-03.jpg`,
    "price": 200,
    "rating": 4,
    "title": `Luxury townhouse`,
    "type": `apartment`
  },
  {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      },
      "name": `Amsterdam`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "avatarUrl": `img/1.png`,
      "id": 3,
      "isPro": true,
      "name": `Angelina`
    },
    "id": 3,
    "images": [`img/1.png`, `img/2.png`],
    "isFavorite": false,
    "isPremium": false,
    "location": {
      "latitude": 52.3909553943508,
      "longitude": 4.929309666406198,
      "zoom": 8
    },
    "maxAdults": 4,
    "previewImage": `img/apartment-02.jpg`,
    "price": 350,
    "rating": 3.4,
    "title": `Beautiful & luxurious studio at great location`,
    "type": `apartment`
  },
  {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      },
      "name": `Amsterdam`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "avatarUrl": `img/1.png`,
      "id": 3,
      "isPro": true,
      "name": `Angelina`
    },
    "id": 4,
    "images": [`img/1.png`, `img/2.png`],
    "isFavorite": false,
    "isPremium": false,
    "location": {
      "latitude": 52.3809553943508,
      "longitude": 4.939309666406198,
      "zoom": 8
    },
    "maxAdults": 4,
    "previewImage": `img/apartment-01.jpg`,
    "price": 500,
    "rating": 4.8,
    "title": `Beautiful & luxurious studio at great location`,
    "type": `apartment`
  }
];

export default OFFERS;
