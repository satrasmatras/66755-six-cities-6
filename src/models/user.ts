interface User {
  email: string,
  password: string
}

interface AuthInfo {
  avatarUrl: string,
  email: string,
  id: number,
  isPro: boolean,
  name: string
}

export default User;
