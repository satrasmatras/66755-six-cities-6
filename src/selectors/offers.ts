import Offer from "../models/offer";
import SortType from "../models/sort-type";

export const getOffersByCity = (offers: Offer[], city: string): Offer[] => {
  return offers.filter((offer) => {
    return offer.city.name === city;
  });
};

const sortOffersByPriceAsc = (o1: Offer, o2: Offer) => o1.price - o2.price;
const sortOffersByPriceDesc = (o1: Offer, o2: Offer) => o2.price - o1.price;
const sortOffersByRatingDesc = (o1: Offer, o2: Offer) => o2.rating - o1.rating;

export const sortOffersBySortType = (offers: Offer[], sortType: SortType): Offer[] => {
  switch (sortType) {
    case SortType.PRICE_DESC:
      return [...offers].sort(sortOffersByPriceDesc);
    case SortType.PRICE_ASC:
      return [...offers].sort(sortOffersByPriceAsc);
    case SortType.TOP_RATED:
      return [...offers].sort(sortOffersByRatingDesc);
    default:
      return offers;
  }
};
