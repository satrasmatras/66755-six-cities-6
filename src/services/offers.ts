import Offer from "../models/offer";

export const getOfferById = (id: number): Offer => {
  return [].find((offer) => offer.id === id);
};

export const adaptOfferFromServer = (offer: any): Offer => {

  const {
    // eslint-disable-next-line camelcase
    preview_image,
    // eslint-disable-next-line camelcase
    is_favorite,
    // eslint-disable-next-line camelcase
    is_premium,
    // eslint-disable-next-line camelcase
    max_adults,
    host
  } = offer;

  const {
    // eslint-disable-next-line camelcase
    is_pro,
    // eslint-disable-next-line camelcase
    avatar_url,
  } = host;

  const newHost = {
    ...host,
    // eslint-disable-next-line camelcase
    isPro: is_pro,
    // eslint-disable-next-line camelcase
    avatarUrl: avatar_url,
  };

  delete newHost.is_pro;
  delete newHost.avatar_url;

  delete offer.preview_image;
  delete offer.is_favorite;
  delete offer.is_premium;
  delete offer.host;
  delete offer.max_adults;

  return {
    ...offer,
    // eslint-disable-next-line camelcase
    previewImage: preview_image,
    // eslint-disable-next-line camelcase
    isFavorite: is_favorite,
    // eslint-disable-next-line camelcase
    isPremium: is_premium,
    // eslint-disable-next-line camelcase
    maxAdults: max_adults,
    host: newHost
  };
};
