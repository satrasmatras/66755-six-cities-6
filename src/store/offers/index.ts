import {offersReducer} from './reducer';

export default offersReducer;
