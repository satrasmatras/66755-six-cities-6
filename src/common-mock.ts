import Offer from "./models/offer";
import Comment from "./models/comment";
import City from "./models/city";
import SortType from "./models/sort-type";
import Routes from "./routes";
import {AuthorizationStatus, LoginPayload} from "./store/user/slice";
import AuthInfo from "./models/auth-info";
import {adaptDataToOffer} from "./adapters/offers";
import {adaptDataToComment} from "./adapters/comments";
import {CommentPost} from "./models/comment-post";
import {adaptDataToAuthInfo} from "./adapters/auth-info";
import {RootState} from "./store";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import {redirectMiddleware} from "./store/redirect";

export const MOCK_OFFER_FROM_API = {
  "bedrooms": 3,
  "city": {
    "location": {
      "latitude": 52.370216,
      "longitude": 4.895168,
      "zoom": 10
    },
    "name": `Amsterdam`
  },
  "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  "host": {
    "avatar_url": `img/1.png`,
    "id": 3,
    "is_pro": true,
    "name": `Angelina`
  },
  "id": 1,
  "images": [`img/1.png`, `img/2.png`],
  "is_favorite": false,
  "is_premium": false,
  "location": {
    "latitude": 52.35514938496378,
    "longitude": 4.673877537499948,
    "zoom": 8
  },
  "max_adults": 4,
  "preview_image": `img/1.png`,
  "price": 120,
  "rating": 4.8,
  "title": `Beautiful & luxurious studio at great location`,
  "type": `apartment`
};
export const MOCK_ADAPTED_OFFER: Offer = adaptDataToOffer(MOCK_OFFER_FROM_API);

export const MOCK_OFFERS_FROM_API = [
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API,
  MOCK_OFFER_FROM_API
].map((offer, i) => ({
  ...offer,
  id: i + 1
}));

export const MOCK_ADAPTED_OFFERS: Offer[] = MOCK_OFFERS_FROM_API.map(adaptDataToOffer);

export const MOCK_COMMENT_FROM_API = {
  "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  "date": `2019-05-08T14:13:56.569Z`,
  "id": 1,
  "rating": 4,
  "user": {
    "avatar_url": `img/1.png`,
    "id": 4,
    "isPro": false,
    "name": `Max`
  }
};

export const MOCK_COMMENTS_FROM_API = [
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API,
  MOCK_COMMENT_FROM_API
].map((comment, i) => ({
  ...comment,
  id: i + 1
}));

export const MOCK_ADAPTED_COMMENT: Comment = adaptDataToComment(MOCK_COMMENT_FROM_API);

export const MOCK_ADAPTED_COMMENTS: Comment[] = MOCK_COMMENTS_FROM_API.map(adaptDataToComment);

export const MOCK_CITY: City = {
  "location": {
    "latitude": 52.370216,
    "longitude": 4.895168,
    "zoom": 10
  },
  "name": `Amsterdam`
};

export const MOCK_SORT_TYPE = SortType.TOP_RATED;

export const MOCK_ROUTE = Routes.FAVORITES;

export const MOCK_AUTH_INFO_FROM_API = {
  "avatar_url": `img/1.png`,
  "email": `Oliver.conner@gmail.com`,
  "id": 1,
  "is_pro": false,
  "name": `Oliver.conner`
};

export const MOCK_ADAPTED_AUTH_INFO: AuthInfo = adaptDataToAuthInfo(MOCK_AUTH_INFO_FROM_API);

export const MOCK_AUTHORIZATION_STATUS = AuthorizationStatus.AUTH;

export const MOCK_EMPTY_ACTION = {
  type: `test`,
  payload: 0
};

export const MOCK_LOGIN_PAYLOAD: LoginPayload = {
  email: MOCK_ADAPTED_AUTH_INFO.email,
  password: `12345`
};

export const MOCK_COMMENT_POST: CommentPost = {
  comment: `Lalalalalalallaallaa`,
  rating: 5,
};

export const MOCK_INITIAL_STATE: RootState = {
  offers: {
    offers: MOCK_ADAPTED_OFFERS,
    isLoading: false,
    sortType: MOCK_SORT_TYPE
  },
  city: {
    city: MOCK_CITY
  },
  map: {
    hoveredOffer: MOCK_ADAPTED_OFFER
  },
  user: {
    authorizationStatus: MOCK_AUTHORIZATION_STATUS,
    authInfo: MOCK_ADAPTED_AUTH_INFO
  },
  offer: {
    offer: MOCK_ADAPTED_OFFER,
    offerIsLoading: false,
    comments: MOCK_ADAPTED_COMMENTS,
    commentsAreLoading: false
  },
  favorites: {
    favorites: MOCK_ADAPTED_OFFERS,
    isLoading: false
  }
};

export const MOCK_STORE = configureStore([thunk, redirectMiddleware])(MOCK_INITIAL_STATE);
