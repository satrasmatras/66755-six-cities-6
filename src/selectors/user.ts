import {RootState} from "../store";
import {createSelector} from "reselect";
import {AuthorizationStatus} from "../store/user/slice";

const selectAuthorizationStatus = (state: RootState) => state.user.authorizationStatus;

export const selectIsAuthorized = createSelector(
  selectAuthorizationStatus,
  (status) => status === AuthorizationStatus.AUTH
);
