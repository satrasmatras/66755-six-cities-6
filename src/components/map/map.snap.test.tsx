import React from "react";
import {render} from "@testing-library/react";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {initialState as userInitialState} from "../../store/user/user";
import thunk from "redux-thunk";
import {initialState as offersInitialState} from "../../store/offers/offers";
import Map from "./map";
import {MOCK_ADAPTED_OFFER, MOCK_ADAPTED_OFFERS, MOCK_CITY} from "../../common-mock";

const mockStore = configureStore([thunk]);
const store = mockStore({
  user: userInitialState,
  offers: offersInitialState
});

it(`Should map render correctly`, () => {
  const {container} = render(
      <Provider store={store}>
        <Router history={browserHistory}>
          <Map
            mainOffer={MOCK_ADAPTED_OFFER}
            offers={MOCK_ADAPTED_OFFERS}
            city={MOCK_CITY}
            className={`map cities__map`}
          />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
