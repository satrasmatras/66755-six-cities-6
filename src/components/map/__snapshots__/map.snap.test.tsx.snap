// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should map render correctly 1`] = `
<div>
  <section
    class="map cities__map leaflet-container leaflet-grab leaflet-touch-drag"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://a.basemaps.cartocdn.com/rastertiles/voyager/10/525/336.png"
              style="width: 256px; height: 256px; left: -237px; top: -134px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin-active.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -162px; top: 18px; z-index: 18;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -162px; top: 18px; z-index: 18;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -162px; top: 18px; z-index: 18;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -162px; top: 18px; z-index: 18;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -162px; top: 18px; z-index: 18;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
